go-controller/pkg/ovn/base_network_controller.go

```diff
@@ -784,3 +784,30 @@ func (bnc *BaseNetworkController) isLocalZoneNode(node *kapi.Node) bool {
 func (bnc *BaseNetworkController) isLayer2Interconnect() bool {
 	return config.OVNKubernetesFeature.EnableInterconnect && bnc.NetInfo.TopologyType() == types.Layer2Topology
 }
+
+func (bnc *BaseNetworkController) nodeZoneClusterChanged(oldNode, newNode *kapi.Node, newNodeIsLocalZone bool) bool {
+	// Check if the annotations have changed. Use network topology and local params to skip unecessary checks
+
+	// NodeIDAnnotationChanged and NodeTransitSwitchPortAddrAnnotationChanged affects local and remote nodes
+	if util.NodeIDAnnotationChanged(oldNode, newNode) {
+		return true
+	}
+
+	if util.NodeTransitSwitchPortAddrAnnotationChanged(oldNode, newNode) {
+		return true
+	}
+
+	// NodeGatewayRouterLRPAddrAnnotationChanged would not affect layer3 secondary network
+	if !bnc.IsSecondary() && util.NodeGatewayRouterLRPAddrAnnotationChanged(oldNode, newNode) {
```

Comment:
This would check can potentially be made only for remote nodes

Reply:
ack. I added the check there.